recipe/build.sh

```diff
@@ -1,132 +1,137 @@
 #!/bin/bash
+set -ex
 
-DEBUG_CMAKE_BUILD_SYSTEM=yes
-declare -a CMAKE_DEBUG_ARGS PYTHON_CMAKE_ARGS VAR_DEPS DEPS_DEFAULTS CMAKE_EXTRA_ARGS
+# CMake FindPNG seems to look in libpng not libpng16
+# https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/FindPNG.cmake#L55
+ln -s $PREFIX/include/libpng16 $PREFIX/include/libpng
 
-# C11 should be what is supported by default
-export CXXFLAGS="$CXXFLAGS -D__STDC_CONSTANT_MACROS"
-export CPPFLAGS="${CPPFLAGS//-std=c++17/-std=c++11}"
-export CXXFLAGS="${CXXFLAGS//-std=c++17/-std=c++11}"
+V4L="1"
 
-if [[ ${DEBUG_CMAKE_BUILD_SYSTEM} == yes ]]; then
-#  CMAKE_DEBUG_ARGS+=("--debug-trycompile")
-#  CMAKE_DEBUG_ARGS+=("-Wdev")
-#  CMAKE_DEBUG_ARGS+=("--debug-output")
-#  CMAKE_DEBUG_ARGS+=("--trace")
-  CMAKE_DEBUG_ARGS+=("-DOPENCV_CMAKE_DEBUG_MESSAGES=1")
+if [[ "${target_platform}" == linux-* ]]; then
+    # Looks like there's a bug in Opencv 3.2.0 for building with FFMPEG
+    # with GCC opencv/issues/8097
+    export CXXFLAGS="$CXXFLAGS -D__STDC_CONSTANT_MACROS"
+    OPENMP="-DWITH_OPENMP=1"
 fi
 
-echo "PYTHON_CMAKE_ARGS="
-echo "${PYTHON_CMAKE_ARGS[@]}"
-
-# Set defaults for dependencies that change across OSes
-# This should match the meta.yaml deps section
-if [[ "$build_variant" == "normal" ]]; then
-  echo "Building normal variant"
```

Comment:
here's where the normal/headless variants are handled

Reply:
@danpetry I put back _normal_ and _headless_ variants but the result is unexpected. If I build locally on osx-arm64 for the normal variant only then will be the result:
```
{
  "libopencv-4.10.0-hceb4623_0.tar.bz2": {
    "recipe": {
      "CONDA_BUILD_SYSROOT": "/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk",
      "build_variant": "normal",
      "c_compiler": "clang",
      "cxx_compiler": "clangxx",
      "numpy": "1.23",
      "target_platform": "osx-arm64"
    }
  },
  "opencv-4.10.0-0.tar.bz2": {
    "recipe": {}
  },
  "py-opencv-4.10.0-py311h392ea8f_0.tar.bz2": {
    "recipe": {
      "build_variant": "normal",
      "target_platform": "osx-arm64"
    }
  }
}
```
It looks fine for a normal variant, right? But pip list command in the test environment of the py-opencv subpackage says:
```
+ pip list
Package                Version
---------------------- -------
numpy                  1.26.4
opencv-python          4.10.0
opencv-python-headless 4.10.0
pip                    24.0
setuptools             69.5.1
wheel                  0.43.0
++ pip list
++ wc -l
++ grep opencv-python
+ test 2 -eq 2
++ pip list
++ grep opencv-python-headless
++ wc -l
+ test 1 -eq 1
+ exit 0
```
That's bad because the **headless** variant `opencv-python-headless` is here too. By default, I would say. I don't get where it was set to be built. By Cmake flags?

Reply:
Even worse. If I build on Prefect the same recipe it fails on unix platforms including osx-arm64 (errrr!):
```
File "/var/folders/k1/30mswbxs7r1g6zwn8y4fyt500000gp/T/abs_97c84g12pn/croot/libopencv_1719910675594/test_tmp/run_test.py", line 2, in <module>

    import cv2

ModuleNotFoundError: No module named 'cv2'
```

Reply:
Hmm, I reread logs and build_variant is set to `0` on linux but should be `5`:
```
Building normal variant

Building headless variant

0
```

Reply:
My bad. There was a removed qt_version. I fixed the if/esle logic